src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/X509ChainPolicy.cs

```diff
@@ -109,8 +128,15 @@ public void Reset()
             _revocationFlag = X509RevocationFlag.ExcludeRoot;
             _verificationFlags = X509VerificationFlags.NoFlag;
             _trustMode = X509ChainTrustMode.System;
-            VerificationTime = DateTime.Now;
+            _verificationTime = DateTime.Now;
+            VerificationTimeIgnored = true;
+
             UrlRetrievalTimeout = TimeSpan.Zero; // default timeout
         }
+
+        public X509ChainPolicy Clone()
+        {
+            return (X509ChainPolicy)MemberwiseClone();
```

Comment:
I want this written by hand.

```suggestion
            X509ChainPolicy clone = new X509ChainPolicy
            {
                DisableCertificateDownloads = DisableCertificateDownloads,
                _revocationMode = _revocationMode,
                _revocationFlag = _revocationFlag,
                _verificationFlags = _verificationFlags,
                _trustMode = _trustMode,
                _verificationTime = _verificationTime,
                VerificationTimeIgnored = VerificationTimeIgnored,
                UrlRetrievalTimeout = UrlRetrievalTimeout,
            };
            
            if (_applicationPolicy?.Count > 0)
            {
                clone.ApplicationPolicy.AddRange(_applicationPolicy);
            }
            
            if (_certificatePolicy?.Count > 0)
            {
                ...
            }
            
            if (_customTrustStore?.Count > 0)
            {
                ...
            }
        }
```

And then in tests, `Assert.NotSame(orig.ApplicationPolicy, clone.ApplicationPolicy);` and `Assert.Equal(orig.ApplicationPolicy, clone.ApplicationPolicy);` (and similar for the other collections)